tests/meta/archive/conftest.py

```diff
@@ -98,3 +106,82 @@ def fixture_full_remote_path(remote_storage_repository: str, remote_path: Path)
 def fixture_full_local_path(local_storage_repository: str, trimmed_local_directory: Path) -> str:
     """Returns the merged local repository and trimmed path."""
     return local_storage_repository + trimmed_local_directory.as_posix()
+
+
+@pytest.fixture(name="archive_store")
+def fixture_archive_store(
+    base_store: Store,
+    helpers: StoreHelpers,
+    sample_id,
+    father_sample_id,
+    mother_sample_id,
+    sample_name,
+) -> Store:
+    """Returns a store with samples for both a DDN customer as well as a non-DDN customer."""
+    customer_ddn: Customer = base_store.add_customer(
+        internal_id="CustWithDDN",
+        invoice_address="Baker Street 221B",
+        invoice_reference="Sherlock Holmes",
+        name="Sherlock Holmes",
+        is_clinical=True,
+        data_archive_location="DDN",
+    )
+    customer_without_ddn: Customer = base_store.add_customer(
+        internal_id="CustWithoutDDN",
+        invoice_address="Wallaby Way 42",
+        invoice_reference="P Sherman",
+        name="P Sherman",
+        is_clinical=False,
+        data_archive_location="PDC",
+    )
+
+    new_samples: List[Sample] = [
+        base_store.add_sample(
+            name=sample_name,
+            sex=Gender.MALE,
+            internal_id=sample_id,
+        ),
+        base_store.add_sample(
+            name="sample_2_with_ddn_customer",
+            sex=Gender.MALE,
+            internal_id=mother_sample_id,
+        ),
+        base_store.add_sample(
+            name="sample_without_ddn_customer",
+            sex=Gender.MALE,
+            internal_id=father_sample_id,
+        ),
+    ]
+    new_samples[0].customer = customer_ddn
+    new_samples[1].customer = customer_ddn
+    new_samples[2].customer = customer_without_ddn
+    external_app = base_store.get_application_by_tag("WGXCUSC000").versions[0]
+    wgs_app = base_store.get_application_by_tag("WGSPCFC030").versions[0]
+    for sample in new_samples:
+        sample.application_version = external_app if "external" in sample.name else wgs_app
+    base_store.session.add(customer_ddn)
+    base_store.session.add(customer_without_ddn)
+    base_store.session.add_all(new_samples)
+    base_store.session.commit()
+    return base_store
+
+
+@pytest.fixture(name="archive_api")
+def fixture_archive_api(
+    populated_housekeeper_api: HousekeeperAPI,
+    ddn_dataflow_api: DDNDataFlowApi,
+    archive_store: Store,
+    spring_file: Path,
+    father_sample_id: str,
+    helpers,
+) -> ArchiveAPI:
+    """Returns an ArchiveAPI with a populated housekeeper store and a DDNDataFlowApi.
+    Also adds /home/ as a prefix for eahc SPRING file for the DDNDataFlowApi to accept them."""
+    for spring_file in populated_housekeeper_api.files(tags=[SequencingFileTag.SPRING]):
```

Comment:
This is flagged as a bug (over writing the spring_file parameter) https://sonarcloud.io/project/issues?resolved=false&types=BUG&pullRequest=2282&id=Clinical-Genomics_cg&open=AYmSB57yg2pb7E_9bx2B